File: Java Programs/lucky_modular_array.java
import java.util.Scanner;

public class Main {

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        // Input array
        String[] inputStr = scanner.nextLine().split(" ");
        int n = inputStr.length;
        
        // Check if input length is less than 4
        if (n < 4) {
            System.out.println("Invalid Input");
            return;
        }
        
        // Parse input array
        int[] arr = new int[n];
        for (int i = 0; i < n; i++) {
            arr[i] = Integer.parseInt(inputStr[i]);
        }

        // Calculate modulo values of all consecutive subarrays of length 4
        int maxMod = Integer.MIN_VALUE;
        int luckyArrayStart = 0;
        boolean multipleLuckyArrays = false;
        for (int i = 0; i <= n - 4; i++) {
            int mod = ((arr[i] * arr[i+1] % 9) * arr[i+2] % 9) * arr[i+3] % 9;
            if (mod > maxMod) {
                maxMod = mod;
                luckyArrayStart = i;
                multipleLuckyArrays = false;
            } else if (mod == maxMod) {
                multipleLuckyArrays = true;
            }
        }
        
        // Output
        if (multipleLuckyArrays) {
            System.out.println("There is more than one lucky arrays");
        } else {
            System.out.println(arr[luckyArrayStart] + " " + arr[luckyArrayStart+1] + " " +
                               arr[luckyArrayStart+2] + " " + arr[luckyArrayStart+3]);
            System.out.println(maxMod);
        }
    }
}
                                                                                                                            

File: Python Programs/lucky_modular_array.py
def triple_modular_exponentiation(a, b, c, m):
    return ((pow(a, b, m) * pow(a, c, m)) % m) * pow(a, c, m) % m

def find_lucky_array(arr):
    n = len(arr)
    
    # Check if array length is less than 4
    if n < 4:
        return "Invalid Input"
    
    max_modulo = -1
    lucky_array = []
    
    # Iterate through all consecutive subarrays of length 4
    for i in range(n - 3):
        subarray = arr[i:i+4]
        sorted_subarray = sorted(subarray)
        modulo_value = triple_modular_exponentiation(sorted_subarray[0], sorted_subarray[1], sorted_subarray[2], sorted_subarray[3])
        
        if modulo_value > max_modulo:
            max_modulo = modulo_value
            lucky_array = subarray
        elif modulo_value == max_modulo and subarray != lucky_array:
            return "There is more than one lucky arrays"
    
    return lucky_array, max_modulo

# Take user input
arr = list(map(int, input().split()))

# Find the lucky array and output the result
result = find_lucky_array(arr)
if isinstance(result, tuple):
    lucky_array, max_modulo = result
    print(*lucky_array)
    print(max_modulo)
else:
    print(result)
                                                                                                                            